From 5a870bd1001c473789aeee39027b33adaea5e64e Mon Sep 17 00:00:00 2001
From: Chris Chiu <chris.chiu@canonical.com>
Date: Mon, 28 Jun 2021 22:16:43 +0800
Subject: [PATCH] rtl8xxxu: disable CONFIG_USB_INTERRUPT_IN_PIPE

The interrupt urb corresponds to the CONFIG_USB_INTERRUPT_IN_PIPE
of the vendor driver, and it's not even enabled in the vendor
driver. This increases lots of unnecessary interrupt overhead
for the driver.

(cherry picked from commit 62aa27949e58fcaa78008e1effec6b5ac26cdeb2)
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
index 158eeb1c012b..06a35ffeabb8 100644
--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
@@ -1762,7 +1762,7 @@ static int rtl8xxxu_identify_chip(struct rtl8xxxu_priv *priv)
 		priv->rx_paths = 1;
 		priv->tx_paths = 1;
 		priv->rtl_chip = RTL8188C;
-		priv->usb_interrupts = 1;
+		priv->usb_interrupts = 0;
 		priv->has_wifi = 1;
 	}
 	hw->wiphy->available_antennas_tx = BIT(priv->tx_paths) - 1;
-- 
2.30.2

